ignite/cmd/generate_typescript_client.go

```diff
@@ -0,0 +1,44 @@
+package ignitecmd
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/ignite/cli/ignite/pkg/cliui"
+	"github.com/ignite/cli/ignite/services/chain"
+)
+
+func NewGenerateTSClient() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "ts-client",
+		Short: "Generate Typescript client for your chain's frontend",
+		RunE:  generateTSClientHandler,
+	}
+	c.Flags().AddFlagSet(flagSetProto3rdParty(""))
+	return c
+}
+
+func generateTSClientHandler(cmd *cobra.Command, args []string) error {
+	session := cliui.New()
+	defer session.Cleanup()
+
+	session.StartSpinner("Generating...")
+
+	c, err := newChainWithHomeFlags(cmd, chain.EnableThirdPartyModuleCodegen())
+	if err != nil {
+		return err
+	}
+
+	cacheStorage, err := newCache(cmd)
+	if err != nil {
+		return err
+	}
+
+	if err := c.Generate(cmd.Context(), cacheStorage, chain.GenerateTSClient()); err != nil {
+		return err
+	}
+
+	session.StopSpinner()
+	session.Println("⛏️  Generated Typescript Client")
```

Comment:
Can we also print the path?

Please check `network chain prepare` cmd for the format.

Reply:
@ilgooz we first need to merge #2510 to do it properly because the paths are know within the `Generate` call and not at this point inside the command handler. I checked a proper way to do it and I think it might be by using the `events.Bus` from the session instance so the chain can use it to print feedback messages. What do you think? Should we first work on the #2510 issue and then add the paths to the `ts-client` and `vuex` commands?

Implementing it without the event bus might not be an elegant solution.

Reply:
Ofc, thanks!